vault/provider.go

```diff
@@ -68,6 +68,30 @@ func Provider() terraform.ResourceProvider {
 				DefaultFunc: schema.EnvDefaultFunc("VAULT_CAPATH", ""),
 				Description: "Path to directory containing CA certificate files to validate the server's certificate.",
 			},
+			"auth_login": {
```

Comment:
This question might be naive -- I'm wondering why the login support was added here rather than as a data source? I'm not saying it "must" be done that way, I'm just curious for my own understanding.

Reply:
Doesn't a data source require that the provider be initialized (i.e. configured) first before anything can be done? I'm going off of #346 however I do realize they were trying to use a `resource` instead of `data`.

side note: should I rename this to `token_provider` like y'all were considering in the previous PR? just semantics but doesn't hurt to ask.

Reply:
A provider is configured prior to any of its data sources or resources being used, but a separate data source can (in some cases) be interpolated in a provider block (ie. you could chain vault providers, but UX wise that is probably undesirable). 

There are unfortunately a lot of nuances and issues with provider schema interpolation outside of just variable usage, so the best bet if you need something in a provider block is to add the reading of it to the provider configuration itself.